horovod/mxnet/__init__.py

```diff
@@ -151,8 +150,3 @@ def broadcast_parameters(params, root_rank=0):
     # Run broadcasts.
     for i, tensor in enumerate(tensors):
         broadcast_(tensor, root_rank, str(i))
-
-    # Make sure tensors pushed to MXNet engine get processed such that all
-    # workers are synced before starting training.
-    for tensor in tensors:
-        tensor.wait_to_read()
```

Comment:
I remember there was a case if one worker received the tensor and started the forward-backward computation but other workers have not yet received the tensors. If this early worker finishes the batch well ahead of other workers there would be some problem of Horovod handling such non-synchronized situation. @tgaddair is this still a limitation of Horovod backend?

Reply:
Does that happen when training with only 1 batch of data?? What error will be printed on such a case?

Reply:
@apeforest I'm not sure I follow the desync situation you are describing. If a worker begins forward/backward earlier than all other workers, it will still submit allreduce operations to the MXNet engine, which will not complete until all other workers submit and launch the matching allreduce ops. This should keep them synchronized.

Reply:
@romerojosh You are right. I think this should be fine.